### PR TITLE
Prevent external redirects. Fixes #1582

### LINKED ISSUE #1582: Open Redirect Vulnerability
1. There is open redirect issue at ( http://localhost:33411/auth/login/?_path=%2F ) .The parameter _path= / leads to the issue.
2.   With reference to the Improve redirector, issue #360 , 
   
   ```
       if '//' not in destination:
                   destination = '%s%s' % (self.server_url(), destination)
   ```

The above 2 lines are not present in the latest version of mail pile I downloaded.When the necessary changes were made, that is the above two lines of code were added to httpd.py and rerun the program again. Mail pile isn't starting up.

### PR DESCRIPTION
**Pull request by BjarniRunar**
Prevent Mailpile's login from becoming an open redirect bouncer for the wider Internet.

### CODE REVIEW

**Comment by BjarniRunar** [comment]
I added a few more security fixes to this... there were more ways to play funny games with the login redirector and HTTPD.

**Comment by BjarniRunar** [comment]
Review window timed out, merging!

File: mailpile/auth.py [hunk 1]
```diff
@@ -121,11 +121,10 @@ def _do_redirect(self):
...
         # These are here to prevent people from abusing this to redirect to
-        # arbitrary URLs on the Internet. The first assertion blocks normal
-        # http:// and https:// URLs. The second is a catch-all that prevents
-        # any path that looks like it starts with a protocol: foo:/stuff/.
-        assert('://' not in path)
-        assert('/' in path.split(':')[0])
+        # arbitrary URLs on the Internet.
+        assert('://' not in path)          # https://, http://, ftp://, ...
+        assert(path[:2] != '//')           # //www.example.com/
+        assert('/' in path.split(':')[0])  # mailto:user@example.com
```
Comment (JocelynDelalande) [inline mailpile/auth.py]: IMHO, relying on *urlparse* does the same job (maybe more), is more explicit, robust & better reviewed :

```
from urlparse import urlparse

url = urlparse(path)
assert(not url.scheme and not url.netloc)
```

My too late cents ;-)

Comment (BjarniRunar) [inline mailpile/auth.py]: That's a good idea, thank you. I'll change this and add to a later PR.

**Pull request merged**